### PR TITLE
AKS Azure Integration

### PR DESCRIPTION
**Pull request by erjosito**
Hey, creating this PR in draft mode. Should I split the solution out of the coach's guides, as I did with the BGP one?

### CODE REVIEW

**Comment by erjosito** [comment]
Hey any progress on this?

**Comment by erjosito** [comment]
Anything on this?

File: 039-AKSForEnterprise/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# What The Hack - AKS Enterprise-Grade
+
+## Introduction
+
+This Hack has an objective studying in depth the integration surface between AKS and different Azure services. Some are optional components (for example monitoring), but others are required to successfully operate Kubernetes clusters on Azure (such as networking or storage).
```
Comment (gfilicetti) [inline 039-AKSForEnterprise/README.md:5]: Let's change this paragraph to  make it a little bit clearer:

The objective of this hack is to dive deeply into all aspects of integration between AKS and various Azure services. We have included challenges for many components that are required to operate a Kubernetes cluster on Azure (eg: networking, storage, etc) as well as some optional ones (eg: monitoring).

File: 039-AKSForEnterprise/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Introduction
+
+This Hack has an objective studying in depth the integration surface between AKS and different Azure services. Some are optional components (for example monitoring), but others are required to successfully operate Kubernetes clusters on Azure (such as networking or storage).
+
+## Learning Objectives
+
+In this hack you will start by deploying a 3-tier application to AKS with very specific networking requirements.
+
+The complexity will quickly evolve towards security and storage, finishing with the last challenge focusing on Arc-enabled kubernetes clusters and arc-enabled data services.
```
Comment (gfilicetti) [inline 039-AKSForEnterprise/README.md:11]: Please capitalize 'kubernetes' and 'arc'

File: 039-AKSForEnterprise/Coach/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+* Most challenges can be solved in multiple ways, all of them correct solutions
+* If there is any concept not clear for everybody, try to make participants explain to each other. Intervene only when no participant has the knowledge
+* **Make sure no one is left behind**
+* Make sure participants have a way to share code, ideally git-based
+* Most challenges involve some level of subscription ownership to create identities or service principals, or for the AAD integration challenge.
+* Leave participants try options even if you know it is not going to work, let them explore on themselves. Stop them only if they consume too much time
+* For each challenge, you can ask the least participative members to describe what has been done and why
+
+> :warning: Note that the code snippets provided here are just an orientation for you as a coach, and might not work 100% in your particular environment. They have been tested, but the rapid nature of Azure CLI versions, Kubernetes, AKS, helm, etc makes it very difficult constantly reviewing them on a regular basis. If you find errors in the code, please send a PR to this repo with the correction.
```
Comment (gfilicetti) [inline 039-AKSForEnterprise/Coach/README.md:18]: Please change this to match the style used in other What The Hacks. We don't use quoted sections. Just use a separate paragraph, as you are doing and prefix with a bolded "NOTE:" prefix, eg:

**NOTE:** The code snippets provided here are just an orientation for you as a coach, and might not work 100% in your particular environment. They have been tested, but the rapid nature of Azure CLI versions, Kubernetes, AKS, helm, etc makes it very difficult constantly reviewing them on a regular basis. If you find errors in the code, please send a PR to this repo with the correction.

**Review by gfilicetti — COMMENTED** [review]
Can you please capitalize the folder name so that it is:
- Student/Resources

Also in Coach/Solutions, I don't think there's a need to have a single folder in there called 'yaml', let's just move all those files to Coach/Solutions. Unless you think in the near future there will we other times of solution files?

File: 039-AKSForEnterprise/Coach/01-containers.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
+# Challenge 1. Containers - Coach's Guide
```
Comment (gfilicetti) [inline 039-AKSForEnterprise/Coach/01-containers.md:1]: Again, make sure that the challenge name here, ie: "Containers", matches what you have in the coach README table of contents.

Also, it should be "Challenge #1: " ... change the period to a semicolon.

This applies to all the other Coach guide files too.

File: 039-AKSForEnterprise/Coach/03-aks_monitoring.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
+# 3. Monitoring - Coach's Guide
```
Comment (gfilicetti) [inline 039-AKSForEnterprise/Coach/03-aks_monitoring.md:1]: Change title to: "Challenge 3: Monitoring - Coach's Guide"

**Review by jrzyshr — CHANGES_REQUESTED** [review]
Hi Jose!
Great work!  Navigation links all look good.

I have some questions here...
In /Coach/Solutions   you have a subfolder called "yaml" that has a collection of Yaml files.  I didn't see where these files are referenced in the coach's guide at all.  Are they?  

In /Coach folder, you have a image_test.azcli file.  What is this?  And where is it referenced?  If it's a solution file, it should go in /Coach/Solutions

Also, what will the final title of the hack be?  The folder is named "AKSForEnterprise", but the home page says "AKS Enterprise Grade".   I'm okay with either, just need to pick one.

File: 039-AKSForEnterprise/Coach/08-arc.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
+# Challenge 8. Arc enabled Kubernetes and Arc enabled data Services - Coach's Guide
```
Comment (gfilicetti) [inline 039-AKSForEnterprise/Coach/08-arc.md:1]: Need to fix some capitalization here:

"Challenge 8: Arc Enabled Kubernetes and Arc Enabled Data Services - Coach's Guide"

**Review by gfilicetti — CHANGES_REQUESTED** [review]
As always @erjosito a very easy review with only a few changes needed. Thank you!

File: 039-AKSForEnterprise/README.md (hunk 1 above)
Comment (erjosito) [reply 039-AKSForEnterprise/README.md:5]: Done!

File: 039-AKSForEnterprise/README.md (hunk 2 above)
Comment (erjosito) [reply 039-AKSForEnterprise/README.md:11]: Done!

File: 039-AKSForEnterprise/Coach/README.md (hunk 3 above)
Comment (erjosito) [reply 039-AKSForEnterprise/Coach/README.md:18]: Done!

File: 039-AKSForEnterprise/Student/01-containers.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+Note the two links at the bottom of the page in the picture above will not work at this stage yet.
+
+## Advanced Challenges (Optional)
+
+- Use an open source database (such as mysql, MariaDB or Postgres)
+- If you used your local Docker installation, complete the challenge using Azure Container Instances
+- If you used Azure Container Instances, complete the challenge using your local Docker installation
+
+## Related Documentation
```
[the remark this answers is not in the document]
Comment (erjosito) [reply 039-AKSForEnterprise/Student/01-containers.md:50]: Done!

File: 039-AKSForEnterprise/Student/02-aks_private.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Challenge 2: AKS Network Integration and Private Clusters
```
[the remark this answers is not in the document]
Comment (erjosito) [reply 039-AKSForEnterprise/Student/02-aks_private.md:1]: Done!

File: 039-AKSForEnterprise/Student/08-arc.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
+# Challenge 8: Arc enabled Kubernetes and Arc enabled data Services
```
[the remark this answers is not in the document]
Comment (erjosito) [reply 039-AKSForEnterprise/Student/08-arc.md:1]: Done, but adding hyphen like "Arc-Enabled"

File: 039-AKSForEnterprise/Coach/01-containers.md (hunk 4 above)
Comment (erjosito) [reply 039-AKSForEnterprise/Coach/01-containers.md:1]: But "Containers" already match, right? Replaced dot with colon!

File: 039-AKSForEnterprise/Coach/08-arc.md (hunk 6 above)
Comment (erjosito) [reply 039-AKSForEnterprise/Coach/08-arc.md:1]: Done! (and added hyphen)

File: 039-AKSForEnterprise/Coach/03-aks_monitoring.md (hunk 5 above)
Comment (erjosito) [reply 039-AKSForEnterprise/Coach/03-aks_monitoring.md:1]: Done!

File: 039-AKSForEnterprise/Student/01-containers.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (erjosito) [reply 039-AKSForEnterprise/Student/01-containers.md:50]: Done!

File: 039-AKSForEnterprise/Student/08-arc.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (erjosito) [reply 039-AKSForEnterprise/Student/08-arc.md:1]: Done!

File: 039-AKSForEnterprise/Student/02-aks_private.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (erjosito) [reply 039-AKSForEnterprise/Student/02-aks_private.md:1]: Done!

**Comment by erjosito** [comment]
Hey @jrzyshr to your questions:
- Some of the YAML files are referenced in the solutions
- I have deleted the file test_images.azcli, its information was already contained in the README files for web and API
- Regarding the name for the lab, I dont really care, both would work for me. I have a slight preference for "AKS Enterprise-Grade", but I can take the other too

**Comment by gfilicetti** [comment]
> Can you please capitalize the folder name so that it is:
> 
> * Student/Resources
> 
> Also in Coach/Solutions, I don't think there's a need to have a single folder in there called 'yaml', let's just move all those files to Coach/Solutions. Unless you think in the near future there will we other times of solution files?

@erjosito Can you change the "resources" folder to be uppercase please? I'm not seeing the change on my end..

Coach/Solutions looks perfect now

**Comment by gfilicetti** [comment]
I'm good to go on this review once you upper case the Resources folder, @erjosito .... great work, once again.

**Comment by erjosito** [comment]
Mmmh I could see it in lower case in my cloned repo, probably a Windows thing. Had to change it to a dummy name, and back to Resources, now it seems to have worked. Thanks for checking!

**Review by jrzyshr — APPROVED** [review]
Thanks for clarifying and making the changes.

I'm good to approve!  For the title...  if you prefer "AKS Enterprise-Grade", then lets go with that!  Just update the folder name to "039-AKSEnterpriseGrade"  (I suppose that's one more change! hah!)

**Review by gfilicetti — APPROVED** [review]
Awesome Jose! Glad to see it all come together!

**Pull request merged**